test/images/nettest/nettest.go

```diff
@@ -241,11 +246,22 @@ func contactOthers(state *State) {
 	if err != nil {
 		log.Fatalf("Unable to create client; error: %v\n", err)
 	}
-	// Double check that worked by getting the server version.
-	if v, err := client.Discovery().ServerVersion(); err != nil {
-		log.Fatalf("Unable to get server version: %v\n", err)
-	} else {
-		log.Printf("Server version: %#v\n", v)
+
+	// Try to get the server version until <timeout>; we use a timeout because
+	// the pod might not have immediate network connectivity.
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(sleepTime) {
+		// Double check that worked by getting the server version.
+		if versionInfo, err = client.Discovery().ServerVersion(); err != nil {
+			log.Printf("Unable to get server version: %v\n", err)
```

Comment:
Nit: I think it would help to append "; retrying" to the message here so that in the case that both branches execute after 1 or 2 retries, we don't get conflicting messages in the logs.

Reply:
Done